MapboxSearch/sdk/src/androidTest/java/com/mapbox/search/CategorySearchIntegrationTest.kt

```diff
@@ -493,6 +495,20 @@ internal class CategorySearchIntegrationTest : BaseTest() {
         }
     }
 
+    @Test
+    fun testConsecutiveRequests() {
+        mockServer.enqueueMultiple(createSuccessfulResponse("sbs_responses/category/successful_incorrect_response.json"), 2)
+
+        val task1 = searchEngine.search(TEST_CATEGORY, CategorySearchOptions(requestDebounce = 1000), EmptySearchCallback)
+
+        val callback = BlockingSearchCallback()
+        val task2 = searchEngine.search(TEST_CATEGORY, CategorySearchOptions(), callback)
+        callback.getResultBlocking()
+
+        assertTrue(task1.isCancelled)
+        assertTrue(task2.isDone)
```

Comment:
Can we add more asserts here?
```
assertFalse(task1.isDone)
assertFalse(task2.isCanceled)
```

Reply:
This functionality is tested in `SearchRequestTaskTest`. 

In `SearchRequestTaskTest` we test only task-related functionality, in particular check that both `isDone` and `isCancelled` can't be `true` at the same time.

In `CategorySearchIntegrationTest` we test category search functionality and assume that other parts work correctly.

So I think we can leave it as is.